addons/sale_stock/models/sale_order.py

```diff
@@ -585,6 +585,17 @@ def _action_launch_stock_rule(self, previous_product_uom_qty=False):
                 line.name, line.order_id.name, line.order_id.company_id, values))
         if procurements:
             self.env['procurement.group'].run(procurements)
+
+        orders = self.mapped('order_id')
+        for order in orders:
```

Comment:
The task mentions reserving product for SO and MO but you only did the SO part. Did we change our mind regarding mrp or did you forget it ?

Reply:
Isn't having the setting on the (manufacturing) Operation Type enough? SOs need extra code to do the auto-reservation, hence the change in `_action_launch_stock_rule` (which is really just a move of the logic from the `procurement_jit` adapted to this new scheme).

Reply:
I think I did not understand the spec correctly. I was thinking that confirming MO could trigger the assign directly but it's only for the MOs created via scheduler, right ?

Reply:
Checking with Thomas... I think I misunderstood the spec too. I assumed it was just loosening the existing auto-reservation logic when an incoming picking (with relevant components) is completed. After all it's easy to push the "Check Availability" button immediately after pushing the "Confirm" button.